src/surfaceqg.jl

```diff
@@ -0,0 +1,370 @@
+  end
+  @. N += vars.Fh
+  return nothing
+end
+
+
+# ----------------
+# Helper functions
+# ----------------
+
+"""
+    updatevars!(prob)
+
+Update variables in `vars` with solution i`n `sol`.
+"""
+function updatevars!(prob)
+  vars, grid, sol = prob.vars, prob.grid, prob.sol
+  @. vars.bh = sol
+  @. vars.uh =   im * grid.l  * sqrt.(grid.invKrsq) * sol
+  @. vars.vh = - im * grid.kr * sqrt.(grid.invKrsq) * sol
+  ldiv!(vars.b, grid.rfftplan, deepcopy(vars.bh))
+  ldiv!(vars.u, grid.rfftplan, deepcopy(vars.uh))
+  ldiv!(vars.v, grid.rfftplan, deepcopy(vars.vh))
+  return nothing
+end
+
+"""
+    set_b!(prob, b)
+
+Set the solution sol as the transform of b and update variables v
+on the grid grid.
+"""
+function set_b!(prob, b)
+  params, vars, grid, sol = prob.params, prob.vars, prob.grid, prob.sol
+  mul!(sol, grid.rfftplan, b)
+  CUDA.@allowscalar sol[1, 1] = 0 # zero domain average
+  updatevars!(prob)
+  return nothing
+end
+
+"""
+    kinetic_energy(prob)
+
+Returns the domain-averaged surface kinetic energy. In SQG flows this is
+identical to half the domain-averaged surface buoyancy variance.
+"""
+@inline function kinetic_energy(prob)
+  sol, vars, grid = prob.sol, prob.vars, prob.grid
+  @. vars.uh =   im * grid.l  * sqrt.(grid.invKrsq) * sol
+  @. vars.vh = - im * grid.kr * sqrt.(grid.invKrsq) * sol
+  @. vars.uh = 0.5 * ( abs2(vars.uh) + abs2(vars.vh) )
+  return 1 / (grid.Lx * grid.Ly) * parsevalsum(vars.uh, grid)
+end
+
+"""
+    buoyancy_variance(prob)
+
+Returns the domain-averaged buoyancy variance. In SQG flows this is identical to
+the domain-averaged velocity variance (twice the kinetic energy)
+"""
+@inline function buoyancy_variance(prob)
+  sol, grid = prob.sol, prob.grid
+  return 1 / (grid.Lx * grid.Ly) * parsevalsum(abs2.(sol), grid)
+end
+
+"""
+    buoyancy_dissipation(prob)
+
+Returns the domain-averaged dissipation rate of surface buoyancy variance due
+to small scale diffusion/viscosity. nν must be >= 1.
+
+In an SQG flow this is identical to twice the rate of kinetic energy dissipation
+"""
+@inline function buoyancy_dissipation(prob)
+  sol, vars, params, grid = prob.sol, prob.vars, prob.params, prob.grid
+  @. vars.uh = 2*grid.Krsq^params.nν * abs2(sol)
+  CUDA.@allowscalar vars.uh[1, 1] = 0
+  return params.ν / (grid.Lx * grid.Ly) * parsevalsum(vars.uh, grid)
+end
+
+"""
+    buoyancy_work(prob)
+    buoyancy_work(sol, v, grid)
+
+Returns the domain-averaged rate of work of buoyancy variance by the forcing Fh.
+"""
+@inline function buoyancy_work(prob)
+  sol, vars, params, grid = prob.sol, prob.vars, prob.params, prob.grid
+  @. vars.uh =  sol * conj(vars.Fh)
+  return 1 / (grid.Lx * grid.Ly) * parsevalsum(vars.uh, grid)
+end
+
+@inline function buoyancy_work(sol, vars::StochasticForcedVars, grid)
+  @. vars.uh =  (vars.prevsol + sol) / 2 * conj(vars.Fh) # Stratonovich
+  # @. vars.uh = grid.invKrsq * vars.prevsol * conj(vars.Fh)           # Ito
```

Comment:
I supposed this commented-out line appears in the `TwoDimTurbulence` module. But I find it confusing --- which is right? Should this line be uncommented sometimes? Removing it may be the best course if we have decided that "Stratonovich" is either correct, or the convention that we are using when calculating budgets. 

@navidcy any comments?

Reply:
Another option is to expand the written comments here to explain when this line should be uncommented.

Reply:
I think we should remove all Ito/Stratonovich remarks and perhaps expand the [docs section](https://fourierflows.github.io/GeophysicalFlowsDocumentation/dev/forcing/) or write up a literated example on this manner.